src/ucs/datastruct/ptr_map.h

```diff
@@ -0,0 +1,37 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifndef UCS_PTR_MAP_H_
+#define UCS_PTR_MAP_H_
+
+#include "khash.h"
+
+#include <ucs/sys/compiler.h>
+#include <ucs/type/status.h>
+
+#include <stdint.h>
+
+BEGIN_C_DECLS
+
+
+#define UCS_PTR_MAP_KEY_INVALID    UINT64_MAX
```

Comment:
can we remove UCS_PTR_MAP_KEY_INVALID ? i see it's not really used for operations

Reply:
it's returned from "put" on error, but OK, will replace with ucs_status_t + 1 out arg

Reply:
can put call ucs_fatal() is failed?

Reply:
what if it happens on server like app? I think all err handling related code has to be handled more graceful

Reply:
IMO it's either allocation failure, or reusing same request pointer which is a bug in UCX, how would there be a graceful error handling?

Reply:
let's assume option 2, I would print ucs_bug() with propagation err code to upper level it would probably leaded to some bug in app (or maybe handled somehow), memory leak and stack some client processes... but even in worst case it maybe better vs crash. On other hand, crash easier to debug.

Reply:
i'd prefer crash, also this is our policy everywhere, we are not paying perf overheads to recover from bugs

Reply:
anyway we have to check against `UCS_KH_PUT_FAILED` and `UCS_KH_PUT_KEY_PRESENT` for return status or ucs_fatal, user can ignore status (for example in "direct" case) or handle error when graceful err handling is preferable.